test/servinge2e/user_permissions_test.go

```diff
@@ -20,118 +27,132 @@ const (
 	helloworldText        = "Hello World!"
 )
 
-func TestUserPermissions(t *testing.T) {
+type allowedOperations struct {
+	get    bool
+	list   bool
+	create bool
+	delete bool
+}
+
+func init() {
+	servingv1.AddToScheme(scheme.Scheme)
+	networkingv1alpha1.AddToScheme(scheme.Scheme)
+	autoscalingv1alpha1.AddToScheme(scheme.Scheme)
+}
 
+func TestUserPermissions(t *testing.T) {
 	caCtx := test.SetupClusterAdmin(t)
```

Comment:
Intentionally leaving out a test for the cluster admin user? If yes, this `caCtx` then appears unused.
Otherwise this stuff looks good to me.

Reply:
Fixed, thanks!